data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconStateHandlerTest.java

```diff
@@ -87,86 +91,67 @@ public void shouldReturnNotFoundWhenQueryAgainstMissingRootObject() throws Excep
 
   @Test
   public void shouldReturnBadRequestWhenNoParameterSpecified() throws Exception {
-    final BeaconStateHandler handler =
-        new BeaconStateHandler(combinedChainDataClient, jsonProvider);
+    final BeaconStateHandler handler = new BeaconStateHandler(provider, jsonProvider);
     when(context.queryParamMap()).thenReturn(Map.of());
-
     handler.handle(context);
-
     verify(context).status(SC_BAD_REQUEST);
   }
 
   @Test
   public void shouldReturnBadRequestWhenBadSlotSpecified() throws Exception {
-    final BeaconStateHandler handler =
-        new BeaconStateHandler(combinedChainDataClient, jsonProvider);
+    final BeaconStateHandler handler = new BeaconStateHandler(provider, jsonProvider);
     when(context.queryParamMap()).thenReturn(Map.of(SLOT, List.of("not-an-int")));
-
     handler.handle(context);
-
     verify(context).status(SC_BAD_REQUEST);
   }
 
   @Test
   public void shouldReturnBadRequestWhenBadParamSpecified() throws Exception {
-    final BeaconStateHandler handler =
-        new BeaconStateHandler(combinedChainDataClient, jsonProvider);
+    final BeaconStateHandler handler = new BeaconStateHandler(provider, jsonProvider);
     when(context.queryParamMap()).thenReturn(Map.of(EPOCH, List.of("not-an-int")));
-
     handler.handle(context);
-
     verify(context).status(SC_BAD_REQUEST);
   }
 
   @Test
   public void shouldReturnBeaconStateObjectWhenQueryByRoot() throws Exception {
-    final BeaconStateHandler handler =
-        new BeaconStateHandler(combinedChainDataClient, jsonProvider);
+    final BeaconStateHandler handler = new BeaconStateHandler(provider, jsonProvider);
     when(context.queryParamMap()).thenReturn(Map.of(ROOT, List.of(blockRoot.toHexString())));
 
     handler.handle(context);
 
     verify(context).result(args.capture());
     SafeFuture<String> data = args.getValue();
-    assertEquals(data.get(), jsonProvider.objectToJSON(beaconState));
+    assertEquals(data.get(), jsonProvider.objectToJSON(new BeaconState(beaconStateInternal)));
```

Comment:
do we need to create a new BeaconState here?

Reply:
nvm I get it. beaconStateInternal is the internal data structure

Reply:
might make it clearer to make it a method call getMeTheExternalRepresentationOfTheInternalState(beaconStateInternal)

Reply:
we're constructing the actual object we can use... i think the bigger problem is where we mix the handling of both objects, then it gets confusing... i'm trying to clean it up, but its not quite right yet...

Reply:
I just realised that object is already constructed above, i'll clean this one up and it'll be more clear...